lib/recurly/base.php

```diff
@@ -112,7 +112,29 @@ protected static function _delete($uri, $client = null)
     return null;
   }
 
+  /**
+   * Returns URI for resource, throwing error if resource code is an empty string
+   * @param string Resource codes and paths
+   * @return string URI
+   * @throws Recurly_Error
+   */
+  public static function _safeUri(...$params) {
+    $uri = '';
+    foreach($params as $param) {
+      // throws error if param is an empty string
+      if (empty(trim($param))) {
+        throw new Recurly_Error("Resource code cannot be an empty value");
+      }
+      $path = '/' . rawurlencode($param);
+      $uri .= $path;
+    }
+    return $uri;
+  }
+
   protected static function _uriWithParams($uri, $params = null) {
+    if ($uri[0] !== '/' && strpos($uri, "https") === false){
```

Comment:
Could this be:

```php
!strpos($uri, "https")
```

Reply:
No, `strpos` find "the position of the first occurrence of a substring in a string" or it returns false. It doesn't return true. I guess this is the closest thing PHP has to `str.include?`.

Reply:
Okay, this is correct then